src/logic/collectibles/store/selectors/index.ts

```diff
@@ -3,26 +3,36 @@ import { createSelector } from 'reselect'
 
 import { NFT_ASSETS_REDUCER_ID, NFT_TOKENS_REDUCER_ID } from 'src/logic/collectibles/store/reducer/collectibles'
 import { safeActiveAssetsSelector } from 'src/routes/safe/store/selectors'
+import { NFTAssets, NFTTokens, NFTAsset } from '../../sources/OpenSea'
+import { GnosisState } from 'src/store'
 
-export const nftAssetsSelector = (state) => state[NFT_ASSETS_REDUCER_ID]
-export const nftTokensSelector = (state) => state[NFT_TOKENS_REDUCER_ID]
+export const nftAssetsSelector = (state: GnosisState): NFTAssets => state[NFT_ASSETS_REDUCER_ID]
+export const nftTokensSelector = (state: GnosisState): NFTTokens => state[NFT_TOKENS_REDUCER_ID]
 
-export const nftAssetsListSelector = createSelector(nftAssetsSelector, (assets) => {
-  return assets ? List(Object.entries(assets).map((item) => item[1])) : List([])
-})
+export const nftAssetsListSelector = createSelector(
+  nftAssetsSelector,
+  (assets): List<NFTAsset> => {
+    return assets ? List(Object.values(assets)) : List()
```

Comment:
What's the point of transforming plain object to immutable.js list?

Reply:
it's not a plain object, it's the object's values (an array) that I'm transforming to a List

Reply:
That's not my point, the point is why are we doing it in the selector?

Reply:
If we use immutable it has to be stored as an immutable.js data structure in redux store

Reply:
but since we're planning to get rid of immutable.js it's fine if you just remove it from there

Reply:
This was still not addressed